bin/geninfo

```diff
@@ -470,14 +476,21 @@ if (system_no_output(3, $gcov_tool, "--help") == -1)
 ($gcov_version, $gcov_version_string) = get_gcov_version();
 $gcov_caps = get_gcov_capabilities();
 
+# Check validity of branch coverage options
+if($br_coverage eq "exclude-exceptions") {
+	$br_coverage = 2;
+} elsif(($br_coverage lt 0) || ($br_coverage gt 1)) {
+	die("ERROR: Invalid lcov_branch_coverage value '$br_coverage'");
+}
+
 # Determine intermediate mode
 if ($rc_intermediate eq "0") {
 	$intermediate = 0;
 } elsif ($rc_intermediate eq "1") {
 	$intermediate = 1;
 } elsif (lc($rc_intermediate) eq "auto") {
 	# Use intermediate format if supported by gcov
-	$intermediate = ($gcov_caps->{'intermediate-format'} ||
+	$intermediate = (($gcov_caps->{'intermediate-format'} && ($br_coverage != 2)) ||
```

Comment:
This case specially modifies the behavior of intermediate=auto in case the user wanted to exclude exception coverage and gcov does provide text intermediate format (which does not provide enough information to perform exception exclusion). I think this is the spirit of the "auto" behavior.

The behavior of intermediate=auto is modified from the previous behavior (only when exception exclusion is requested by the user) in the following way:
- If gcov supports intermediate text format, then non-intermediate processing is automatically selected.

All the other casuistics remain the same.

I don't consider that the user might be confused by this behavior, since the program has automatically chosen the best option, which is what the user requested.

If the program aborts with an error, the user will have to modify his lcovrc or pass explicitly an option to set intermediate=0, which seems pretty silly given that we've got the "auto" option.